algorithms/linfa-ftrl/src/lib.rs

```diff
@@ -4,22 +4,15 @@ mod algorithm;
 mod error;
 mod hyperparams;
 
-use crate::hyperparams::FtrlParams;
+use crate::hyperparams::{FtrlParams, FtrlValidParams};
 pub use algorithm::Result;
 pub use error::FtrlError;
 use linfa::Float;
 use ndarray::Array1;
 use ndarray_rand::RandomExt;
 use rand::distributions::Uniform;
 
-#[cfg(feature = "serde")]
-use serde_crate::{Deserialize, Serialize};
-
-#[cfg_attr(
-    feature = "serde",
-    derive(Serialize, Deserialize),
-    serde(crate = "serde_crate")
-)]
+#[derive(Clone, Debug)]
```

Comment:
I'd still derive `serde` on these types, since we'te including `Serde` unconditionally

Reply:
What I meant in this comment is that this you should add `Serialize, Deserialize` to this derive, and delete the `cfg_attr` above, since it depends on an non-existent `serde` feature. Do this for all types with `serde` annotations.

Reply:
Oh, I got it wrong, apologies!